instrumentation/aws_sdk/lib/opentelemetry/instrumentation/aws_sdk/instrumentation.rb

```diff
@@ -43,10 +44,28 @@ def require_dependencies
           require_relative 'handler'
           require_relative 'message_attributes'
           require_relative 'messaging_helper'
+          require_relative 'patches/telemetry'
         end
 
         def add_plugins(*targets)
-          targets.each { |klass| klass.add_plugin(AwsSdk::Plugin) }
+          targets.each do |klass|
+            next if supports_telemetry_plugin?(klass)
+
+            klass.add_plugin(AwsSdk::Plugin)
+          end
+        end
+
+        def supports_telemetry_plugin?(klass)
+          telemetry_plugin? &&
+            klass.plugins.include?(Aws::Plugins::Telemetry)
+        end
+
+        def telemetry_plugin?
+          ::Aws.const_defined?('Plugins::Telemetry')
+        end
+
+        def patch
```

Comment:
Might be good to have a slightly more descriptive name like `patch_telemtry_plugin`.  And some documentation on what it does/why